### PR TITLE
Tweak GitHub Actions

### PR DESCRIPTION
**Pull request by cosmo0920**
* Drop Ruby 2.4 task
* Add Ruby 3.0 and ruby-head tasks

Signed-off-by: Hiroshi Hatake <hatake@calyptia.com>

### CODE REVIEW

**Comment by cosmo0920** [comment]
@ashie @kenhys @okkez Could you kindly take a look?

File: .github/workflows/windows.yml [hunk 1]
```diff
@@ -5,12 +5,19 @@ on:
 jobs:
   build:
     runs-on: ${{ matrix.os }}
+    continue-on-error: ${{ matrix.experimental }}
     strategy:
       fail-fast: false
       matrix:
         ruby: [ '2.4', '2.5', '2.6', '2.7' ]
```
Comment (ashie) [inline .github/workflows/windows.yml:12]: Don't you update it?

**Review by ashie — CHANGES_REQUESTED** [review]

File: .github/workflows/windows.yml (hunk 1 above)
Comment (cosmo0920) [reply .github/workflows/windows.yml]: Fixed in https://github.com/fluent/fluent-plugin-rewrite-tag-filter/pull/111/commits/eeca8ba0e7ee03c1e92a3ce1e8278717856c0a66.

**Comment by cosmo0920** [comment]
Thanks!

**Pull request merged**